source/frontend/src/features/properties/map/project/add/ProductsArrayForm.tsx

```diff
@@ -0,0 +1,132 @@
+import { LinkButton, StyledRemoveLinkButton } from 'components/common/buttons';
+import GenericModal from 'components/common/GenericModal';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import { Section } from 'features/mapSideBar/tabs/Section';
+import { FieldArray, FieldArrayRenderProps, FormikProps, useFormikContext } from 'formik';
+import { useProductProvider } from 'hooks/providers/useProductProvider';
+import React, { useRef, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaTrash } from 'react-icons/fa';
+import styled from 'styled-components';
+
+import { ProductForm, ProjectForm } from './models';
+import ProductSubForm from './ProductSubForm';
+
+export interface IProductsArrayFormProps {
+  field: string;
+  formikProps: FormikProps<ProjectForm>;
+}
+
+export const ProductsArrayForm: React.FunctionComponent<IProductsArrayFormProps> = ({
+  field,
+  formikProps,
+}) => {
+  const arrayHelpersRef = useRef<FieldArrayRenderProps | null>(null);
+  const { values } = useFormikContext<ProjectForm>();
+
+  const [showModal, setShowModal] = useState(false);
+  const [showFileModal, setShowFileModal] = useState(false);
+  const [rowToDelete, setRowToDelete] = useState<number | undefined>(undefined);
+
+  const { retrieveProductFile, retrieveProductFileLoading } = useProductProvider();
+
+  const products = values.products || [];
+
+  const handleRemove = async (index: number) => {
+    const productId = products[index].id;
+    if (productId !== undefined && productId !== 0) {
+      const file = await retrieveProductFile(productId);
+      if (file !== undefined && file !== null) {
+        setShowFileModal(true);
+      } else {
+        showDeleteWarning(index);
+      }
+    } else {
+      showDeleteWarning(index);
+    }
+  };
+
+  const showDeleteWarning = (index: number) => {
+    setShowModal(true);
+    setRowToDelete(index);
+  };
+
+  const confirmDelete = () => {
+    if (rowToDelete !== undefined) {
```

Comment:
hmm, I feel like this is one of those weird error states that should never occur but if it does occur its hard to know what to do. Do we need a warning to the user here or is that a waste?

Reply:
IMO it would be a waste. The only chance this happens is if the state setting didnt happen (got to confirm without seen the prompt first)